src/LibraryWindow.vala

```diff
@@ -1260,4 +1279,106 @@ public class Music.LibraryWindow : LibraryWindowInterface, Hdy.ApplicationWindow
 
         return base.configure_event (event);
     }
+
+    private class ImportErrorToast : Granite.Widgets.Toast {
+        public Gee.HashMultiMap<Gst.PbUtils.DiscovererResult, string> import_errors { get; construct; }
+
+        public ImportErrorToast (string title) {
+            base (title);
+        }
+
+        construct {
+            import_errors = new Gee.HashMultiMap<Gst.PbUtils.DiscovererResult, string> ();
+            set_default_action (_("Details"));
+
+            default_action.connect (() => {
+                var invalid_files = import_errors.@get (Gst.PbUtils.DiscovererResult.URI_INVALID);
+                var error_files = import_errors.@get (Gst.PbUtils.DiscovererResult.ERROR);
+                var timeout_files = import_errors.@get (Gst.PbUtils.DiscovererResult.TIMEOUT);
+                var busy_files = import_errors.@get (Gst.PbUtils.DiscovererResult.BUSY);
+                var missing_plugin_files = import_errors.@get (Gst.PbUtils.DiscovererResult.MISSING_PLUGINS);
+                var music_folder = Settings.Main.get_default ().music_folder;
+
+                var primary_text = _("Issues while importing from %s").printf (music_folder);
+
+                string secondary_text = "";
+                string invalid_text = "";
+                string error_text = "";
+                string timeout_text = "";
+                string busy_text = "";
+                string missing_plugin_text = "";
+
+                if (invalid_files.size > 0) {
+                    invalid_text = _("Invalid uri: %i").printf (invalid_files.size) + "\n";
+                }
+
+                if (error_files.size > 0) {
+                    error_text = _("Could not be read and may be damaged: %i").printf (error_files.size) + "\n";
+                }
+
+                if (timeout_files.size > 0) {
+                    timeout_text = _("Took too long to load: %i").printf (timeout_files.size) + "\n";
+                }
+
+                if (busy_files.size > 0) {
+                    busy_text = _("Already being used: %i").printf (busy_files.size) + "\n";
+                }
+
+                if (missing_plugin_files.size > 0) {
+                    missing_plugin_text = _("Required gstreamer plugin not installed: %i").printf (missing_plugin_files.size) + "\n";
```

Comment:
Gstreamer is a niche technical detail that probably doesn't mean anything to most people. Mentioning "codecs" might be a little more generally understandable, but maybe there's a better word for this too. Maybe something in the direction of "Additional software codecs are needed:"